src/connector/src/sink/elasticsearch_opensearch/elasticsearch_opensearch_formatter.rs

```diff
@@ -147,7 +151,27 @@ impl ElasticSearchOpenSearchFormatter {
             match op {
                 Op::Insert | Op::UpdateInsert => {
                     let key = self.key_encoder.encode(rows)?;
-                    let value = self.value_encoder.encode(rows)?;
+                    let mut col_indices = match self.value_encoder.col_indices() {
+                        Some(col_indices) => col_indices.to_vec(),
+                        None => (0..self.value_encoder.schema().len()).collect(),
+                    };
+                    if let Some(delete_row) = update_map.remove(&key) {
+                        delete_row
+                            .iter()
+                            .enumerate()
+                            .zip_eq_debug(rows.iter())
+                            .for_each(|((index, insert_column), delete_column)| {
+                                if let Some(insert_column) = insert_column
+                                    && let Some(delete_column) = delete_column
+                                    && insert_column == delete_column
+                                {
+                                    col_indices.retain(|&x| x != index);
```

Comment:
It is more efficient to populate the new `col_indices` proactively instead of calling `retain` on the full list for each column. Example:
```
if let Some(insert_column) = insert_column
        && let Some(delete_column) = delete_column
        && insert_column == delete_column {
  // do nothing
} else {
  modified_col_indices.push_back(index);
}
```